src/main/kotlin/club/sk1er/vigilance/Vigilant.kt

```diff
@@ -16,7 +16,10 @@ import kotlin.reflect.jvm.javaField
 abstract class Vigilant @JvmOverloads constructor(
     file: File,
     val guiTitle: String = "Settings",
-    private val propertyCollector: PropertyCollector = JVMAnnotationPropertyCollector()
+    private val propertyCollector: PropertyCollector = JVMAnnotationPropertyCollector(),
+    var categoryComparator: Comparator<in Category> = compareBy { it.name },
+    var subcategoryComparator: Comparator<in Map.Entry<String, List<PropertyData>>> = compareBy { null },
+    var propertyComparator: Comparator<in PropertyData> = compareBy { it.attributes.subcategory }
```

Comment:
Not sure why these properties are `var`s, and I'm also not sure I like how complex the types are. Can we extract this behavior to an interface or abstract class developers can implement for custom behavior, and make this constructor accept only one extra parameter?

Reply:
Is it now ok?
I cannot really do anything to the `Comparator<in Map.Entry<String, List<PropertyData>>>` type, because there is no proper Subcategory class to use for that. I also wouldn't want to create one just for this, because it feels too unnecessary.